### PR TITLE
Best practices for backup: strage and storage media

### PR DESCRIPTION
**Pull request by core-man**
Referring to https://core-man.github.io/blog/post/backup/#%E7%BD%91%E7%9B%98%E5%A4%87%E4%BB%BD

Related to #43.

### CODE REVIEW

**Comment by core-man** [comment]
I am thinking if we need to discuss file-system here, like https://core-man.github.io/blog/post/backup/#%E6%96%87%E4%BB%B6%E7%B3%BB%E7%BB%9F.

We have to choose a format for the portable disk, right?

File: source/best-practices/backup.rst [hunk 1]
```diff
@@ -1,61 +1,133 @@
...
 ----------
 
-- 硬盘备份
-- 网盘备份
+我们需要一定的存储空间来备份电脑里的文件，常用的备份介质有\ **硬盘**\ 和\ **网盘**\ 。
+
+最常见的备份介质是移动硬盘，其特点是携带方便、价格便宜。近几年，移动硬盘发展迅速，市场上
+常见的存储空间大小有 4TB 和 5TB，基本可以满足日常科研的备份需求。如果不够那就买两块吧。
+一些研究组可能也会购买存储空间更大的磁盘阵列，一般由许多块几 TB 大小的硬盘组成，价格较贵。
+将磁盘阵列挂载到本地电脑后，使用起来就像一块超大的移动硬盘一样，非常方便。
```
Comment (seisman) [inline source/best-practices/backup.rst:43]:
```suggestion
最常见的备份介质是移动硬盘，其特点是携带方便、价格便宜。近几年，移动硬盘发展迅速，市场上
常见的存储空间大小有 4TB 和 5TB，基本可以满足日常科研的备份需求。
```

File: source/best-practices/backup.rst [hunk 2]
```diff
@@ -1,61 +1,133 @@
...
+- `百度网盘 <https://pan.baidu.com>`__\ ：全平台网盘，免费空间可以有 2 TB。
+- `Dropbox <https://www.dropbox.com/>`__\ ：国外最好用的全平台网盘，熟练掌握
+  科学上网技巧的人可以使用。免费用户只有 2 GB 容量，最大可扩容到 18 GB 以上。
+  付费用户容量为 1 TB。
+- `MEGA <https://mega.io/>`__\ ：全平台网盘，免费空间 50 GB。
+- `Google Drive <https://www.google.com/drive/>`__\ ：Google 云端硬盘。免费容量 15 GB，由 Google 云端网盘、
+  `Google 相册 <https://www.google.com/photos/>`__ \ 、Gmail 共用。
+- `OneDrive <https://www.microsoft.com/en-us/microsoft-365/onedrive/online-cloud-storage>`__\ ：
+  微软的云存储服务，初始免费容量 5 GB。Word、Excel、PowerPoint 以及 Onenote 等微软软件的文件
+  都可以直接保存到 OneDrive 中。
```
Comment (seisman) [inline source/best-practices/backup.rst:63]: Better to show a comparison using a table.

The information we may be interested:

1. Free quota and price for more quota
2. Cross-platform? Linux? macOS? Windows?
3. Easy to use in China?

File: source/best-practices/backup.rst [hunk 3]
```diff
@@ -1,61 +1,133 @@
...
+
+----
+
 **备份！备份！备份！**
 
 尽管不经常发生，但电脑随时可能会坏掉或被盗，电脑硬盘也随时可能会出问题，
-也可能不小心使用一个命令把几个月的工作误删了。所以，备份非常重要，不进行必要的备份
-是十分危险的。
+也可能不小心使用一个命令（如 ``rm *``）把几个月的工作误删了。所以，备份非常重要，不进行必要的备份
+是十分危险的。笔者和一些同行多次体验过数据误删或硬盘损坏又未备份带来的痛苦。
```
Comment (seisman) [inline source/best-practices/backup.rst:15]: I think @CovMat mentioned that the average lifetimes of hard drives are about 5 years, due to the frequent disk I/O in scientific research. I think that's a big warning for everyone.

File: source/best-practices/backup.rst [hunk 4]
```diff
@@ -1,61 +1,133 @@
...
 - 全盘备份
 
+不同备份策略的主要区别是所需的硬盘空间和备份时间。例如，若仅备份一些重要文件，
+则需要的硬盘空间较短，备份所需要的时间也较小。若要全盘备份，则需要较大的硬盘空间和
+较长的备份时间。仅备份家目录介于二者之间。
+
+建议用户根据个人需求和存储空间来决定具体的备份策略。如果某些文件非常重要，
+丢失后恢复难度极大或恢复时间极长，强烈推荐备份一下，如毕业论文、无法再次获取的数据、
+经过长时间处理得到的资料。如果拥有很大的硬盘存储空间，可以进一步选择备份家目录，
+甚至全盘备份。
```
Comment (seisman) [inline source/best-practices/backup.rst:33]: It would be better to list the pros and cons of each backup scheme. Maybe using a table?

**Comment by seisman** [comment]
> I am thinking if we need to add a new section to discuss file format, like https://core-man.github.io/blog/post/backup/#%E6%96%87%E4%BB%B6%E7%B3%BB%E7%BB%9F.
> 
> Or we can choose to include it under backup media?
> 
> We have to choose a format for the portable disk, right?

The original post is too long to read. I think we may summarize to a few sentences.

**Comment by core-man** [comment]
> > I am thinking if we need to add a new section to discuss file format, like https://core-man.github.io/blog/post/backup/#%E6%96%87%E4%BB%B6%E7%B3%BB%E7%BB%9F.
> > Or we can choose to include it under backup media?
> > We have to choose a format for the portable disk, right?
> 
> The original post is too long to read. I think we may summarize to a few sentences.

Yes, we shall make it a few sentences. I mean shall we add a new subsection or just include it in a subsection?

```
硬盘格式
^^^^^^^^
```

File: source/best-practices/backup.rst [hunk 5]
```diff
@@ -1,61 +1,131 @@
...
+  微软的云存储服务，初始免费容量 5 GB。Word、Excel、PowerPoint 以及 Onenote 等微软软件的文件
+  都可以直接保存到 OneDrive 中。
+
+用户可以根据自己的需求（即是同步文件还是备份文件）和资金，选择适合的网盘。国内用户
+推荐使用坚果云在不同设备之间进行文件传输和同步。
+
+以下我们主要讨论如何使用移动硬盘来备份文件。
 
 Linux
 ------
```
Comment (seisman) [inline source/best-practices/backup.rst:70]: Can we just add "Linux 下移动硬盘应选择 XXX 格式" here?

Comment (core-man) [reply source/best-practices/backup.rst:70]: ok

File: source/best-practices/backup.rst (hunk 2 above)
Comment (core-man) [reply source/best-practices/backup.rst]: done in cde691d.

File: source/best-practices/backup.rst (hunk 3 above)
Comment (core-man) [reply source/best-practices/backup.rst:19]: done in b345fe6.

File: source/best-practices/backup.rst (hunk 4 above)
Comment (core-man) [reply source/best-practices/backup.rst]: see d23dcc5.

**Comment by core-man** [comment]
I am wondering if we need to move the online disk to a single rst cause #103 also needs it.

**Comment by seisman** [comment]
> I am wondering if we need to move the online disk to a single rst cause #103 also needs it.
> 
> Or we may just link it to the location in this chapter when we are working on file transfering.

We can make the decision later.

File: source/best-practices/backup.rst [hunk 6]
```diff
@@ -1,61 +1,158 @@
...
+.. table:: 不同备份策略的比较
+   :align: center
+
+   ================ ====================================================== ===============
+   备份的策略         优点                                                   缺点
+   ================ ====================================================== ===============
+   仅备份重要文件     所需硬盘空间较小、所需备份时间较短                          仅重要文件备份了，若硬盘摔坏，其他文件需要重新获取
+   仅备份家目录       科研和日常文件基本在家目录下，硬盘摔坏带来的损失很小          所需硬盘空间较大、所需备份时间较长
+   全盘备份          硬盘摔坏带来的损失几乎为零                                 所需硬盘空间最大、所需备份时间最长
+   ================ ====================================================== ===============
```
Comment (seisman) [inline source/best-practices/backup.rst:46]: Sorry, the table looks bad. Need to find a better way.

File: source/best-practices/backup.rst [hunk 7]
```diff
@@ -1,61 +1,158 @@
...
+
+建议用户根据个人需求和存储空间来决定具体的备份策略。如果\ **文件非常重要**\ ，
+丢失后\ **恢复难度极大**\ 或\ **恢复时间极长**\ ，强烈推荐备份一下，如毕业论文、脚本、代码、
+无法再次获取的数据、经过长时间处理得到的资料。如果拥有更多硬盘存储空间且备份时间能够接受，
+可以进一步选择备份家目录，甚至全盘备份。
```
Comment (seisman) [inline source/best-practices/backup.rst:51]: Considering that external hard drivers are so cheap now. Perhaps we should always recommend users to backup the home directory?

Comment (core-man) [reply source/best-practices/backup.rst:51]: Yes to me.

Comment (seisman) [reply source/best-practices/backup.rst:51]: Then, we may even not need to mention netdisk backup.

Comment (core-man) [reply source/best-practices/backup.rst:51]: Yes, netdisk is mainly used to sync files among different devices, right?

Comment (seisman) [reply source/best-practices/backup.rst:51]: Yes to me.

Comment (core-man) [reply source/best-practices/backup.rst:51]: removed.

Comment (core-man) [reply source/best-practices/backup.rst]: See update in 3c27bc9.

File: source/best-practices/backup.rst [hunk 8]
```diff
@@ -1,61 +1,103 @@
...
+   本章正在编写中，尚未完成。
+
+----
+
 **备份！备份！备份！**
 
 尽管不经常发生，但电脑随时可能会坏掉或被盗，电脑硬盘也随时可能会出问题，
-也可能不小心使用一个命令把几个月的工作误删了。所以，备份非常重要，不进行必要的备份
-是十分危险的。
+也可能不小心使用一个命令（如 ``rm *``）把几个月的工作误删了。所以，备份非常重要，不进行必要的备份
```
Comment (seisman) [inline source/best-practices/backup.rst:18]:
```suggestion
也可能不小心使用一个命令（如 ``rm -r *``）把几个月的工作误删了。所以，备份非常重要，不进行必要的备份
```

File: source/best-practices/backup.rst [hunk 9]
```diff
@@ -1,61 +1,103 @@
...
+
+.. table:: 不同备份策略的比较
+   :align: center
+
+   ================ ====================================================== ===============
+   备份的策略         优点                                                   缺点
+   ================ ====================================================== ===============
+   备份家目录        科研和日常文件基本在家目录下，硬盘摔坏带来的损失很小           所需硬盘空间较大、所需备份时间较长
+   全盘备份          硬盘摔坏带来的损失几乎为零                                 所需硬盘空间最大、所需备份时间最长
+   ================ ====================================================== ===============
```
Comment (seisman) [inline source/best-practices/backup.rst:44]: Difficult to read on mobile phones, and it seems useless to compare these two schemes.

Comment (core-man) [reply source/best-practices/backup.rst:44]: I am revising it.

Comment (core-man) [reply source/best-practices/backup.rst]: Please check 5ace09827c30eba99e909aedd159e15551807a53

File: source/best-practices/backup.rst [hunk 10]
```diff
@@ -1,61 +1,92 @@
...
 尽管不经常发生，但电脑随时可能会坏掉或被盗，电脑硬盘也随时可能会出问题，
-也可能不小心使用一个命令把几个月的工作误删了。所以，备份非常重要，不进行必要的备份
-是十分危险的。
+也可能不小心使用一个命令（如 ``rm -r *``）把几个月的工作误删了。所以，备份非常重要，不进行必要的备份
+是十分危险的。笔者和一些同行多次体验过数据误删或硬盘损坏又未备份带来的痛苦。
+
+.. warning::
 
-备份的目的
+   日常科研工作中，电脑硬盘经常进行大量的读写操作，直接影响硬盘的寿命。一般而言，电脑硬盘的平均
```
Comment (seisman) [inline source/best-practices/backup.rst:26]:
```suggestion
   日常科研工作中，电脑硬盘经常进行大量的读写操作，直接影响硬盘的寿命。我们的经验表明，电脑硬盘的平均
```

File: source/best-practices/backup.rst [hunk 11]
```diff
@@ -1,61 +1,92 @@
...
+   寿命大约 5 年左右。当然，硬盘的品质和使用情况不同，其寿命也有所不同。
```
Comment (seisman) [inline source/best-practices/backup.rst:27]:
```suggestion
   寿命大约为 5 年。当然，硬盘的品质和使用情况不同，其寿命也有所不同。
```

File: source/best-practices/backup.rst [hunk 12]
```diff
@@ -1,61 +1,89 @@
...
-- 网盘备份
+备份的策略简单概括有两种：
+
+- 备份家目录或其部分子目录
+- 全盘备份
 
-我们主要讨论硬盘备份。
+不同备份策略的主要区别是所需的硬盘空间和备份时间，以及硬盘摔坏带来的损失。
+移动硬盘是最常见的备份设备，其特点是携带方便、价格便宜。近几年，移动硬盘发展迅速，
+市场上常见的存储空间大小有 4TB 和 5TB，单价不到 1000 块，基本可以满足日常科研的备份需求。
```
Comment (seisman) [inline source/best-practices/backup.rst:39]: List the average prices for 1-5 TB hard drivers?

Comment (core-man) [reply source/best-practices/backup.rst]: List a range in 646fe8a.

File: source/best-practices/backup.rst [hunk 13]
```diff
@@ -1,61 +1,90 @@
...
 ^^^^^^
 
-至少每周将工作电脑中的文件备份到移动硬盘中。推荐使用 rsync 进行备份，其用法为::
+至少每周将工作电脑中的文件备份到移动硬盘中。推荐使用 ``rsync`` 命令进行备份，其用法为::
 
-    rsync --delete -av /home/seisman/ /data1/seisman/
+    $ rsync --delete -av /home/seismo-learn/ /data1/seismo-learn/
 
-该命令作用是将 /home/seisman/ 目录完整同步到 /data1/seisman/ 目录下。
+该命令作用是将 /home/seismo-learn/ 目录完整同步到 /data1/seismo-learn/ 目录下。
```
Comment (seisman) [inline source/best-practices/backup.rst:55]: 1. Change `/home/seismo-learn/` to <code>:file:`/home/seismo-learn`</code>
2. Are you sure that we need the trailing slash (i.e., `/home/seismo-learn/`, not `/home/seismo-learn`
3. If yes to 2, better to add a warning that the trailing slash is important
4. Change `/data1` to a more reasonable name `/mnt/xxx`?

File: source/best-practices/backup.rst [hunk 14]
```diff
@@ -1,61 +1,90 @@
 备份与还原
 ==========
 
-**备份！备份！备份！**
+:本节贡献者: |田冬冬|\（作者）、
```
Comment (seisman) [inline source/best-practices/backup.rst:4]: Change the author order andal date

File: source/best-practices/backup.rst (hunk 13 above)
Comment (core-man) [reply source/best-practices/backup.rst:55]: We will work on it in another PR, right? I just revise this part a little.

We better prepare a script`backup.sh`

Comment (core-man) [reply source/best-practices/backup.rst]: > 1. Change `/home/seismo-learn/` to `` :file:`/home/seismo-learn` ``
done

> 2. Are you sure that we need the trailing slash (i.e., `/home/seismo-learn/`, not `/home/seismo-learn`
> 3. If yes to 2, better to add a warning that the trailing slash is important

We will work on this part in another PR?

> 4. Change `/data1` to a more reasonable name `/mnt/xxx`?
done

Comment (seisman) [reply source/best-practices/backup.rst]: OK

**Review by seisman — APPROVED** [review]

**Pull request merged**